src/models/util/Lasso.ts

```diff
@@ -0,0 +1,51 @@
+function CrossesLine(
+  start: [number, number],
+  end: [number, number],
+  p: [number, number]
+): boolean {
+  if (start[1] > end[1]) {
+    if (!(p[1] <= start[1] && p[1] > end[1])) {
+      return false;
+    }
+  } else {
+    if (!(p[1] <= end[1] && p[1] > start[1])) {
+      return false;
+    }
+  }
+  if (
+    p[0] >=
+    ((end[0] - start[0]) * (p[1] - start[1])) / (end[1] - start[1]) + start[0]
+  ) {
+    return true;
+  }
+  return false;
```

Comment:
nit: can return this conditional
```suggestion
  return p[0] >= ((end[0] - start[0]) * (p[1] - start[1])) / (end[1] - start[1]) + start[0]
```

Reply:
Can you add a comment explaining the math of this?

Reply:
And spoil the fun?  I'll add a link to the wikipedia article on even-odd rule.